challenge_4/python/shaowen/bi_tree_reverse.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jan  4 11:01:42 2017
+
+@author: shaowen Liu
+"""
+
+class Node():
+    def __init__(self, data, left_child = None, right_child = None):
+        self.data = data
+        self.left = left_child
+        self.right = right_child
+    
+    def has_child(self):
+        return (self.left or self.right)
+    
+def bi_insert(root, new_node):
```

Comment:
You may want to specify either in documentation or through an error checking procedure that `new_node` is indeed a `Node` object. I know you yourself wouldn't make the mistake of trying to add a number rather than an object, but it's a thing to take into consideration for others!